src/server/protocol.ts

```diff
@@ -442,6 +442,12 @@ namespace ts.server.protocol {
           * Errorcodes we want to get the fixes for.
           */
         errorCodes?: number[];
+
+        /**
+         * An optional set of settings to be used when computing indentation.
+         * If argument is omitted - then it will use settings for file that were previously set via 'configure' request or global settings.
+         */
+        options?: EditorSettings;

```

Comment:
@mhegazy On my machine, it appeared that with a freshly opened editor, the "default" settings tsserver recieved on the first request might not agree with user/workspace settings set in VS Code.

It looks like the vscode extension has a cache in place to track whether or not they already sent formatting options [here](https://github.com/Microsoft/vscode/blob/master/extensions/typescript/src/features/formattingProvider.ts#L76), so in principle `ensureFormatOptions` could be exposed to the quickfix provider. We might additionally need a method `getFormatOptionsOrUndefined` which is a getter for the cache.